### DISCUSSION TITLE
Standard way to handle singleton (i.e. table with single row)?

### DISCUSSION THREAD

**Discussion by chrisjsewell** [sqlalchemy/sqlalchemy#11645] [Usage Questions]
Heya,

I have some singular project metadata I want to store in the database (an sqlite database to be precise), e.g.

```python
@dataclass
class Metadata:
   version: str
   description: str
   ...
```

I guess this would be a table with a single row 🤷

Is there a "standard" pattern for doing this in sqlalchemy? (in particular, using the `@mapped_as_dataclass` approach)

----

edit: obviously I know the alternative would be to do something like:

```python
@REGISTRY.mapped_as_dataclass
class ProjectMetadata:
    """A table to store singular metadata about the project."""

    __tablename__ = "project_metadata"

    id: Mapped[intpk] = mapped_column(init=False)
    key: Mapped[str] = mapped_column(unique=True)
    value: Mapped[str]
```

but then you lose knowledge of the actual allowed keys, and all values have to be the same type

**Comment by CaselIT** [comment]
Hi,

sqlalchemy doesn't define anything in particular for this case.
Of your option one is normalized by column, meaning that any time you add a new property you need an `alter table` statement, while the other is a plain key-value table. (I would avoid the id there, you can use `key` as primary key).

If you want you can define key as an enum, so that the values are prederined.
